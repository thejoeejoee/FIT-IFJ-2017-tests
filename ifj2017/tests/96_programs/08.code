' Draw primary numbers from interval
declare function prvocislo(cislo as integer) as boolean

function prvocislo(a as integer) as boolean
   
    dim delitel as integer = a - 1
    
    do while delitel > 1
    
        if ((a \ delitel) * delitel) = a then
            return false
        else
        end if
        delitel = delitel - 1
    loop
    
   
    
    return true
end function

scope
    dim flag as boolean
    print !"Enter two positive integers:\n";
    
    dim n1 as integer
    dim n2 as integer
    
    input n1
    print !"\n";
    input n2
    print !"\n";
    
    print !"Prime numbers between"; n1 ; !" and"; n2 ; !" are";
    
    dim i as integer = n1 + 1
    
    do while i < n2
    
        flag = prvocislo(i)
        
        if flag then
            print i;
        else
        end if
        
        i = i + 1
    loop

   
end scope
