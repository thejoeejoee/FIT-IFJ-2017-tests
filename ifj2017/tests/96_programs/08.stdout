Enter two positive integers:
? 
? 
Prime numbers between 10 and 20 are 11 13 17 19